### ISSUE TITLE
SSH Remote cannot install on the host: Hanging when downloading with wget

### ISSUE THREAD

**Issue by thanhleha-kit** [microsoft/vscode-remote-release#2582]
My remote is an Ubuntu 18.04 desktop version with NFS file system. I managed to use the extension with the same remote and local before. Then at some point it doesn't work anymore. I tried to remove all extensions, reinstall vscode, set up the configuration file one again, downgrade the extension (from 0.50.0 to 0.49.0). All stop at "Downloading with wget" and hanging there for hours (which should be done in 1-2 minutes). 

I checked the remote folder (`~/vscode-server/bin/fe22a9645b44368865c0ba92e2fb881ff1afce94/`), deleted files (lock files) in there, deleted the whole folder, nothing works. The `vscode-server.tar.gz` is there. Apparently, the remote installation downloaded the file with wget successfully but doesn't know how to proceed.  

To access to my remote, I need VPN. But I can use vscode SSH Remote with that VPN before, and I can also ssh to my remote from any kind of terminals. 

- VSCode Version: 1.43.1 (Standard) and 1.44.0-insider
- Local OS Version: Windows 10 Pro (10.0.18363)
- Remote OS Version: Ubuntu 18.04
- Remote Extension/Connection Type: SSH

Steps to Reproduce:

1. Connect to the Remote
2. Look into OUTPUT of Remote - SSH (from the dropdownlist)
  
The full log:
[21:09:47.291] Log Level: 2
[21:09:47.297] remote-ssh@0.50.0
[21:09:47.298] win32 x64
[21:09:47.300] SSH Resolver called for "ssh-remote+myremotepc", attempt 1
[21:09:47.301] SSH Resolver called for host: myremotepc
[21:09:47.301] Setting up SSH remote "myremotepc"
[21:09:47.304] Acquiring local install lock: C:\Users\yyyyyy\AppData\Local\Temp\vscode-remote-ssh-myremotepc-install.lock
[21:09:47.339] Looking for existing server data file at c:\Users\yyyyyy\AppData\Roaming\Code\User\globalStorage\ms-vscode-remote.remote-ssh\vscode-ssh-host-mypc-fe22a9645b44368865c0ba92e2fb881ff1afce94\data.json
[21:09:47.340] Using commit id "fe22a9645b44368865c0ba92e2fb881ff1afce94" and quality "stable" for server
[21:09:47.342] Install and start server if needed
[21:09:47.376] Checking ssh with "ssh -V"
[21:09:47.435] > OpenSSH_for_Windows_7.7p1, LibreSSL 2.6.5
[21:09:47.435] Checking ssh with "C:\WINDOWS\System32\OpenSSH\ssh.exe -V"
[21:09:47.482] > OpenSSH_for_Windows_7.7p1, LibreSSL 2.6.5
[21:09:47.483] Checking ssh with "C:\Program Files\Git\usr\bin\ssh.exe -V"
[21:09:47.535] > OpenSSH_7.9p1, OpenSSL 1.1.1a  20 Nov 2018
[21:09:47.537] askpass server listening on \\.\pipe\vscode-ssh-askpass-8897286afde542f11b034b173b98aaf36ce0422c-sock
[21:09:47.538] Spawning local server with {"ipcHandlePath":"\\\\.\\pipe\\vscode-ssh-askpass-f8352aad9a4d24ac57b6fa837f5f39e79bde8e79-sock","sshCommand":"C:\\Program Files\\Git\\usr\\bin\\ssh.exe","sshArgs":["-v","-T","-D","59797","myremotepc"],"dataFilePath":"c:\\Users\\yyyyyy\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-vscode-remote.remote-ssh\\vscode-ssh-host-myremotepc-fe22a9645b44368865c0ba92e2fb881ff1afce94\\data.json"}
[21:09:47.538] Local server env: {"DISPLAY":"1","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"c:\\Users\\yyyyyy\\.vscode\\extensions\\ms-vscode-remote.remote-ssh-0.50.0\\out\\local-server\\askpass.bat","VSCODE_SSH_ASKPASS_NODE":"C:\\Program Files\\Microsoft VS Code\\Code.exe","VSCODE_SSH_ASKPASS_MAIN":"c:\\Users\\yyyyyy\\.vscode\\extensions\\ms-vscode-remote.remote-ssh-0.50.0\\out\\askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"\\\\.\\pipe\\vscode-ssh-askpass-8897286afde542f11b034b173b98aaf36ce0422c-sock"}
[21:09:48.005] Spawned 6432
[21:09:48.113] > local-server> Spawned ssh: 1192
[21:09:48.155] stderr> OpenSSH_7.9p1, OpenSSL 1.1.1a  20 Nov 2018
[21:09:48.326] stderr> debug1: Server host key: ecdsa-sha2-nistp256 .......
[21:09:48.573] stderr> Authenticated to myremotepc.somedomain ([myremotepcip]:22).
[21:09:48.888] > Welcome to Ubuntu 18.04.4 LTS (GNU/Linux 5.3.0-42-generic x86_64)
> 
>  * Documentation:  https://help.ubuntu.com
>  * Management:     https://landscape.canonical.com
>  * Support:        https://ubuntu.com/advantage
> 
>  * Latest Kubernetes 1.18 beta is now available for your laptop, NUC, cloud
>    instance or Raspberry Pi, with automatic updates to the final GA release.
> 
>      sudo snap install microk8s --channel=1.18/beta --classic
> 
>  * Multipass 1.1 adds proxy support for developers behind enterprise
>    firewalls. Rapid prototyping for cloud operations just got easier.
> 
>      https://multipass.run/
> 
>  * Canonical Livepatch is available for installation.
>    - Reduce system reboots and improve kernel security. Activate at:
>      https://ubuntu.com/livepatch
> 
> 0 packages can be updated.
> 0 updates are security updates.
> 
> Your Hardware Enablement Stack (HWE) is supported until April 2023.
[21:09:48.923] stderr> /home/xxxxx/.bashrc: line 63: bind: warning: line editing not enabled
[21:09:48.923] stderr> /home/xxxxx/.bashrc: line 64: bind: warning: line editing not enabled
[21:09:49.598] > ready: 17a308b52147
[21:09:49.632] > Linux 5.3.0-42-generic #34~18.04.1-Ubuntu SMP Fri Feb 28 13:42:26 UTC 2020
[21:09:49.634] Platform: linux
[21:09:49.678] > 17a308b52147: running
[21:09:49.697] > Acquiring lock on /home/xxxxx/.vscode-server/bin/fe22a9645b44368865c0ba92e2fb881ff1afce94/vscode-remote-lock.xxxxx.fe22a9645b44368865c0ba92e2fb881ff1afce94
[21:09:49.710] > \ln /home/xxxxx/.vscode-server/bin/fe22a9645b44368865c0ba92e2fb881ff1afce94/vscode-remote-lock.xxxxx.fe22a9645b44368865c0ba92e2fb881ff1afce94.target /home/xxxxx/.vscode-server/bin/fe22a9645b44368865c0ba92e2fb881ff1afce94/vscode-remote-lock.xxxxx.fe22a9645b44368865c0ba92e2fb881ff1afce94
> Installing to /home/xxxxx/.vscode-server/bin/fe22a9645b44368865c0ba92e2fb881ff1afce94...
> 17a308b52147%%1%%
> Downloading with wget

**Comment by zhmwkaka** [comment]
Same issue here. Have you resolved it?

VSCode: 1.43.1
Remote SSH Extension: 0.50.1
Remote OS: Centos 7
Local OS: Windows 10 Home 19041.153

**Comment by thanhleha-kit** [comment]
> Same issue here. Have you resolved it?
> 
> VSCode: 1.43.1
> Remote SSH Extension: 0.50.1
> Remote OS: Centos 7
> Local OS: Windows 10 Home 19041.153

Unfortunately, not.

**Comment by thanhleha-kit** [comment]
> Same issue here. Have you resolved it?
> 
> VSCode: 1.43.1
> Remote SSH Extension: 0.50.1
> Remote OS: Centos 7
> Local OS: Windows 10 Home 19041.153

Now it works like charm (after upgrading from 0.50.0 to 0.50.1). Can you try it again?

**Comment by roblourens** [comment]
Hm, did you try deleting the downloaded .tar? Does it look like it is done, can you untar it? 

Could you try with VS Code Insiders, to see whether there is an issue with that particular binary on the server?

Does the server connect to the internet through a proxy?

**Comment by thanhleha-kit** [comment]
> Hm, did you try deleting the downloaded .tar? Does it look like it is done, can you untar it?
> 
> Could you try with VS Code Insiders, to see whether there is an issue with that particular binary on the server?
> 
> Does the server connect to the internet through a proxy?

I specified those information in the OP:
- I tried deleting the tar, tried deleting the whole folder, also tried untar-ing it (and successfully). But nothing worked before. I also used VSCode Insiders (v1.44.0)  but it was the same problem. I tried restarting and reinstalling my vscode. The problem remained the same. But now, after several days, it works.
- I need a VPN to reach the remote. But before that the extension had worked, and now it works with the VPN.

Only one thing I could think of. 2 days before the problem happened, our remote file system (nfs) was moved. Maybe it was not stable enough for installing the extension on the remote? But we didn't remount or do anything with it and now it works. Just wait for a few days. My Remote WSL, on the other hand, works as usual.
